src/app/pages/system/general-settings/support/file-ticket-form/file-ticket-form.component.ts

```diff
@@ -0,0 +1,224 @@
+      const exceedLimitFile = fileList && [...fileList].some((file) => file.size >= this.FILE_SIZE_LIMIT_50Mb);
+      if (exceedLimitFile) {
+        this.fileList = null;
+        this.form.patchValue({ screenshot: null });
+        this.form.get('screenshot').setErrors({
+          ixManualValidateError: {
+            message: this.translate.instant('File size is limited to 50 MiB.'),
+          },
+        });
+        this.cdr.markForCheck();
+      } else {
+        this.form.get('screenshot').setErrors(null);
+        this.cdr.markForCheck();
+      }
+    });
+  }
+
+  getCategories(): Observable<Option[]> {
+    return this.form.get('token').value$.pipe(
+      filter((token) => !!token),
+      debounceTime(300),
+      switchMap((token) => this.ws.call('support.fetch_categories', [token])),
+      map((choices) => {
+        let options: Option[] = [];
+        for (const property in choices) {
+          if (choices.hasOwnProperty(property)) {
+            options.push({ label: property, value: choices[property] });
+          }
+        }
+        options = _.sortBy(options, ['label']);
+        return options;
+      }),
+    );
+  }
+
+  restoreToken(): void {
+    const token = this.sysGeneralService.getTokenForJira();
+    if (token) {
+      this.form.patchValue({ token });
+    }
+  }
+
+  onSubmit(): void {
+    const values = this.form.value;
+
+    const payload = {
+      category: values.category,
+      title: values.title,
+      body: values.body,
+      type: values.type,
+      token: values.token,
+    } as CreateNewTicket;
+
+    if (values.attach_debug) {
+      payload.attach_debug = values.attach_debug;
+    }
+
+    this.isFormLoading = true;
+
+    this.openDialog(payload);
+  }
+
+  openDialog(payload: CreateNewTicket): void {
+    const dialogRef = this.matDialog.open(EntityJobComponent, {
+      data: {
+        title: this.translate.instant('Ticket'),
+        closeOnClickOutside: true,
+      },
+    });
+    dialogRef.componentInstance.setCall('support.new_ticket', [payload]);
+    dialogRef.componentInstance.submit();
+    dialogRef.componentInstance.success.pipe(
+      untilDestroyed(this),
+    ).subscribe((res: Job<NewTicketResponse>) => {
+      let ticket: NewTicketResponse;
+      if (res.result) {
+        ticket = res.result;
+      }
+      if (res.method === 'support.new_ticket' && this.fileList?.length) {
+        for (const file of Array.from(this.fileList)) {
+          const formData: FormData = new FormData();
+          formData.append('data', JSON.stringify({
+            method: 'support.attach_ticket',
+            params: [{
+              ticket: res.result.ticket,
+              filename: file.name,
+              token: payload.token,
+            }],
+          }));
+          formData.append('file', file);
+          dialogRef.componentInstance.wspost(this.apiEndPoint, formData);
+          dialogRef.componentInstance.success.pipe(untilDestroyed(this)).subscribe(() => {
```

Comment:
1. I've attached two files to my ticket and none of them appeared in Jira. I've waited for 7 minutes.
2. We should not be uploading images silently. I may be uploading a large image, see the success dialog, think everything is done and close the browser. See if you can rewrite the code to show user a dialog. Also, would be good to use switchMaps, not go deeper into callbacks.